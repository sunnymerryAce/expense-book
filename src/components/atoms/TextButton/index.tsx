import React from "react";
import Button from "@material-ui/core/Button";
import useStyles from "./style";

export interface TextButtonProps {
  text: string;
  handleClick: (...props: any[]) => any;
}

const TextButton: React.FC<TextButtonProps> = (props) => {
  const classes = useStyles();
  return (
    <Button
      variant="outlined"
      color="primary"
      size="large"
      className={`TextButton ${classes.root}`}
      onClick={props.handleClick}
    >
      {props.text}
    </Button>
  );
};

export default TextButton;
