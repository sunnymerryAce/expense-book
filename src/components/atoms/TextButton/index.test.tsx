import React from "react";
import TextButton from "./";
import { shallow } from "enzyme";

describe("<TextButton />", () => {
  let container: any = null;
  const props = {
    text: "buttonText",
    handleClick: jest.fn(),
  };

  beforeEach(() => {
    container = shallow(<TextButton {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    expect(container.find(".TextButton").props()).toMatchObject({
      onClick: props.handleClick,
    });
  });

  test("should show proper text", () => {
    expect(container.find(".TextButton").text()).toBe(props.text);
  });

  test("event handler should be triggered", () => {
    container.simulate("click");
    expect(props.handleClick).toBeCalled();
  });
});
