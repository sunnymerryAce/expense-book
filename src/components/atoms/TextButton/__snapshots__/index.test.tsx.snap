// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined TextButton makeStyles-root-1 MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">buttonText</span></button>"`;
